dpc/src/circuits/inner_circuit.rs

```diff
@@ -412,17 +417,25 @@ pub fn execute_inner_circuit<C: Parameters, CS: ConstraintSystem<C::InnerScalarF
             let commitment_cs = &mut cs.ns(|| "Check that record is well-formed");
 
             // Perform noop safety checks.
-            given_value.conditional_enforce_equal(
-                &mut commitment_cs.ns(|| format!("If the input record {} is a noop, enforce it has a value of 0", i)),
-                &zero_value,
-                &given_is_dummy,
-            )?;
-            given_payload.conditional_enforce_equal(
-                &mut commitment_cs
-                    .ns(|| format!("If the input record {} is a noop, enforce it has an empty payload", i)),
-                &empty_payload,
-                &given_is_dummy,
-            )?;
+            {
+                let given_value_field_elements = given_value
+                    .to_constraint_field(&mut commitment_cs.ns(|| "convert given value to field elements"))?;
+                let given_payload_field_elements = given_payload
+                    .to_constraint_field(&mut commitment_cs.ns(|| "convert given payload to field elements"))?;
+
+                given_value_field_elements.conditional_enforce_equal(
+                    &mut commitment_cs
+                        .ns(|| format!("If the input record {} is a noop, enforce it has a value of 0", i)),
+                    &zero_value_field_elements,
+                    &given_is_dummy,
+                )?;
+                given_payload_field_elements.conditional_enforce_equal(
+                    &mut commitment_cs
+                        .ns(|| format!("If the input record {} is a noop, enforce it has an empty payload", i)),
+                    &empty_payload_field_elements,
+                    &given_is_dummy,
```

Comment:
Note: This part is a small change being piggybacked in this PR. @Pratyush mentioned that it is cheaper to compare the field elements that the bytes are packed into, rather than the bytes themselves.